pyfr/plugins/integrate.py

```diff
@@ -52,22 +52,45 @@ def __init__(self, intg, cfgsect, suffix=None):
             # Open
             self.outf = init_csv(self.cfg, cfgsect, ','.join(header))
 
+        # Whether to use a quadrature different to that of the solution points
+        quad = self.cfg.hasopt(cfgsect, 'quad-deg')
```

Comment:
Try to avoid using hasopt, it usually leads to convoluted code.  Try/except based solutions are very often cleaner here.